itest/rust/src/builtin_tests/containers/dictionary_test.rs

```diff
@@ -383,6 +383,24 @@ fn dictionary_iter() {
     assert_eq!(map, map2);
 }
 
+#[itest]
+fn dictionary_iter_size() {
+    let dictionary = dict! {
+        "foo": 0,
+        "bar": true,
+        "baz": "foobar",
+        "nil": Variant::nil(),
+    };
+    let mut iter = dictionary.iter_shared();
+    assert_eq!(iter.size_hint(), (4, Some(4)));
+    iter.next();
+    iter.next();
+    iter.next();
+    assert_eq!(iter.size_hint(), (1, Some(1)));
+    iter.next();
+    assert_eq!(iter.size_hint(), (0, Some(0)));
```

Comment:
Could you also add a case with `Dictionary::new()` -- an empty dictionary?

Reply:
Gladly

Reply:
Added test to cover empty dicts.